### PR TITLE
add connection string api for namespace

### PR DESCRIPTION
**Pull request by ivywei0125**
What does this implement/fix? Explain your changes.
---------------------------------------------------
…

Does this close any currently open issues?
------------------------------------------

Any relevant logs, screenshots, error output, etc.?
-------------------------------------

Any other comments?
-------------------
…

Has this been tested?
---------------------------
- [ ] Tested

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-eventhubs-lib/src/main/java/com/microsoft/azure/toolkit/lib/eventhubs/EventHubsNamespace.java [hunk 1]
```diff
@@ -47,4 +54,33 @@ public String loadStatus(@Nonnull EventHubNamespace remote) {
     public List<EventHubsInstance> getInstances() {
         return this.instanceModule.list();
     }
+
+    public String getOrCreateListenConnectionString() {
+        final List<AccessRights> accessRights = Collections.singletonList(AccessRights.LISTEN);
+        final List<EventHubNamespaceAuthorizationRule> connectionStrings = Optional.ofNullable(getRemote())
+                .map(eventHubInstance -> eventHubInstance.listAuthorizationRules().stream()
+                        .filter(rule -> new HashSet<>(rule.rights()).containsAll(accessRights))
```
Comment (ivywei0125) [inline azure-toolkit-libs/azure-toolkit-eventhubs-lib/src/main/java/com/microsoft/azure/toolkit/lib/eventhubs/EventHubsNamespace.java:62]: contains --> equals

File: azure-toolkit-libs/azure-toolkit-eventhubs-lib/src/main/java/com/microsoft/azure/toolkit/lib/eventhubs/EventHubsNamespace.java [hunk 2]
```diff
@@ -47,4 +54,33 @@ public String loadStatus(@Nonnull EventHubNamespace remote) {
...
+                        .collect(Collectors.toList()))
+                .orElse(new ArrayList<>());
+        if (connectionStrings.size() > 0) {
+            return connectionStrings.get(0).getKeys().primaryConnectionString();
+        }
+        final EventHubsManager manager = getParent().getRemote();
+        if (Objects.isNull(manager)) {
```
Comment (ivywei0125) [inline azure-toolkit-libs/azure-toolkit-eventhubs-lib/src/main/java/com/microsoft/azure/toolkit/lib/eventhubs/EventHubsNamespace.java:69]: call this.exist()

File: azure-toolkit-libs/azure-toolkit-eventhubs-lib/src/main/java/com/microsoft/azure/toolkit/lib/eventhubs/EventHubsNamespace.java [hunk 3]
```diff
@@ -47,4 +54,33 @@ public String loadStatus(@Nonnull EventHubNamespace remote) {
...
+            throw new AzureToolkitRuntimeException(AzureString.format("resource ({0}) not found", getName()).toString());
+        }
+        final String accessRightsStr = StringUtils.join(accessRights, "-");
+        final EventHubNamespaceAuthorizationRule.DefinitionStages.WithAccessPolicy policy = manager.namespaceAuthorizationRules()
+                .define(String.format("policy-%s-%s", accessRightsStr, Utils.getTimestamp()))
```
Comment (ivywei0125) [inline azure-toolkit-libs/azure-toolkit-eventhubs-lib/src/main/java/com/microsoft/azure/toolkit/lib/eventhubs/EventHubsNamespace.java:74]: add azure-toolkit-intellij

File: azure-toolkit-libs/azure-toolkit-eventhubs-lib/src/main/java/com/microsoft/azure/toolkit/lib/eventhubs/EventHubsNamespace.java [hunk 4]
```diff
@@ -47,4 +54,33 @@ public String loadStatus(@Nonnull EventHubNamespace remote) {
...
+                .withExistingNamespace(getResourceGroupName(), getName());
+        EventHubNamespaceAuthorizationRule.DefinitionStages.WithCreate withCreate = policy.withListenAccess();
+        if (accessRights.contains(AccessRights.MANAGE)) {
+            withCreate = policy.withManageAccess();
+        } else if (accessRights.contains(AccessRights.SEND) && accessRights.contains(AccessRights.LISTEN)) {
+            withCreate = policy.withSendAndListenAccess();
+        } else if (accessRights.contains(AccessRights.SEND)) {
+            withCreate = policy.withSendAccess();
+        }
+        return withCreate.create().getKeys().primaryConnectionString();
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-eventhubs-lib/src/main/java/com/microsoft/azure/toolkit/lib/eventhubs/EventHubsNamespace.java:84]: safer to get or create connection string with only `LISTEN` access right

**Review by wangmingliang-ms — APPROVED** [review]
LGTM, remember to resolve comments in later PRs.

**Pull request merged**